packages/trader/src/App/Components/Elements/Modals/ServicesErrorModal/__tests__/services-error-modal.spec.tsx

```
@@ -0,0 +1,90 @@
+                onConfirm={onConfirm}
+            />
+        );
+        expect(container).toBeEmptyDOMElement();
+    });
+
+    test('AuthorizationRequiredModal should render when code is AuthorizationRequired', () => {
+        const services_error_mock = {
+            code: 'AuthorizationRequired',
+            message: 'AuthorizationRequired',
+        };
+        const authorization_props = {
+            is_logged_in: true,
+            is_visible: true,
+            onConfirm: jest.fn(),
+        };
+        render(<ServicesErrorModal services_error={services_error_mock} {...authorization_props} />);
+        expect(screen.getByText('AuthorizationRequiredModal')).toBeInTheDocument();
+    });
+
+    test('InsufficientBalanceModal should render when code is InsufficientBalance', () => {
+        const services_error_mock = {
+            code: 'InsufficientBalance',
+            message: 'InsufficientBalance',
+        };
+        const insufficient_balance_props = {
+            is_virtual: true,
+            is_visible: true,
+            onConfirm: jest.fn(),
+        };
+        render(<ServicesErrorModal services_error={services_error_mock} {...insufficient_balance_props} />);
+        expect(screen.getByText('InsufficientBalanceModal')).toBeInTheDocument();
+    });
+
+    test('CompanyWideLimitExceededModal should render when code is CompanyWideLimitExceeded', () => {
+        const services_error_mock = {
+            code: 'CompanyWideLimitExceeded',
+            message: 'CompanyWideLimitExceeded',
+        };
+        const limit_exceeded_props = {
+            is_visible: true,
+            onConfirm: jest.fn(),
+        };
+        render(<ServicesErrorModal services_error={services_error_mock} {...limit_exceeded_props} />);
+        expect(screen.getByText('CompanyWideLimitExceededModal')).toBeInTheDocument();
+    });
+
+    test('CompanyWideLimitExceededModal should render when code is CompanyWideLimitExceeded', () => {
```

isn't this description duplicated with the line 64
it should be related to mock data as you are passing error props.
```suggestion
    test('Default case should render when code is CompanyWideLimitExceeded', () => {
```

---

Agree